docs/interfaces/query-lifecycle.md

```diff
@@ -0,0 +1,102 @@
+* **Account Decoder**: The account decoder from the [`auth`](.../spec/auth) module, which translates `[]byte`s into accounts.
+* **RPC Client**: The [Tendermint RPC Client](https://github.com/tendermint/tendermint/blob/master/rpc/client/interface.go), or node, to which the request will be relayed to.
+* **Keybase**: A [Key Manager](.//core/accounts-keys.md) used to sign transactions and handle other operations with keys.
+* **Output Writer**: A [Writer](https://golang.org/pkg/io/#Writer) used to output the response.
+* **Configurations**: The flags configured by the user for this command, including `--height`, specifying the height of the blockchain to query and `--indent`, which indicates to add an indent to the JSON response.
+
+For full specification of the `CLIContext` type, click [here](https://github.com/cosmos/cosmos-sdk/blob/master/client/context/context.go#L36-L59).
+
+### Parameters and Route Creation
+
+The next step is to parse the command or request, extract the arguments, create a `queryRoute`, and encode everything.
+
+In this case, `Query` contains a `delegatorAddress` as its only argument. However, the request can only contain `[]byte`s, as it will be relayed to a consensus engine node that has no inherent knowledge of the application types. Thus, the `CLIContext` `codec` is used to marshal the address as the type `QueryDelegatorParams`. All query arguments (e.g. the `staking` module also has `QueryValidatorParams` and `QueryBondsParams`) have their own types that the application `codec` understands how to encode and decode. The module [`querier`](.//building-modules/querier.md) declares these types and the application registers the `codec`s.
+
+A `route` is also created for `Query` so that the application will understand which module to route the query to. Baseapp will understand this query to be a `custom` query in the module `staking` with the type `QueryDelegatorDelegations`. Thus, the route will be `"custom/staking/delegatorDelegations"`.
+
+### ABCI Query
+
+The `CLIContext`'s main `Query` function takes the `route`, which is now called `path`, and arguments, now called `key`. It first retrieves the RPC Client (called the **node**) configured by the user to relay this query to, and creates the `ABCIQueryOptions` (parameters formatted for the ABCI call). The node is then used to make the ABCI call, `ABCIQueryWithOptions`.
+
+## Tendermint and ABCI
+
+With a call to `ABCIQueryWithOptions()`, `Query` arrives at the consensus engine portion of its lifecycle. Nodes running the consensus engine (e.g. Tendermint Core) make ABCI calls to interact with the application. At this point, `Query` exists as an ABCI `RequestQuery` and the [ABCI Client](https://github.com/tendermint/tendermint/blob/51b3428f5c0f4fdd2e469147cd90353faa4bd704/abci/client/client.go#L16-L50) calls the ABCI method [`Query()`](https://tendermint.com/docs/spec/abci/abci.html#query) on the application.
```

Comment:
"the consenssus engine portion" -> not clear enough. 

Say that interfaces (CLI/REST) need to indicate which full node they interact with when they're first configured, or as a flag (+ give an example). Then you can say that the query is first received by the consensus engine of said full node, but you need to explain why.

Reply:
It is important to explain that we use the RPC of the consensus engine, but that queries are not actually part of consensus. They are not processed by all nodes, only by the node that the CLI points to